### PR TITLE
JAMES-3436 Email/set create should support specific headers

### PR DESCRIPTION
**Pull request by chibenwa**
This enables for instance to request read receipts

The following restrictions applies:

 - There MUST NOT be two properties that represent the same header field
    (e.g., header:from and from) within the Email or particular EmailBodyPart.
 - Header fields MUST NOT be specified in parsed forms that are forbidden for that particular field.
 - Header fields beginning with Content- MUST NOT be specified on the Email object, only on EmailBodyPart objects.

### CODE REVIEW

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/EmailSetSerializer.scala [hunk 1]
```diff
@@ -260,7 +262,117 @@ class EmailSetSerializer @Inject()(messageIdFactory: MessageId.Factory, mailboxI
     readMapEntry[ClientPartId, ClientEmailBodyValue](s => Id.validate(s).fold(e => Left(e.getMessage), partId => Right(ClientPartId(partId))),
       clientEmailBodyValueReads)
 
-  private implicit val emailCreationRequestReads: Reads[EmailCreationRequest] = Json.reads[EmailCreationRequest]
+  case class EmailCreationRequestWithoutHeaders(mailboxIds: MailboxIds,
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/EmailSetSerializer.scala:265]: Wouldn't it be better to move that one in `EmailSet` scala file instead?

Comment (chibenwa) [reply server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/EmailSetSerializer.scala:265]: It is only used for deserialization (and avoiding the JsonPath pattern) hence I located it here...

**Comment by chibenwa** [comment]
test this please

**Review by Arsnael — APPROVED** [review]

**Review by LanKhuat — APPROVED** [review]

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**